File: app/src/main/java/com/android/gubonny/simplegithub/api/GithubApi.java
package com.android.gubonny.simplegithub.api;

import com.android.gubonny.simplegithub.api.model.GithubRepo;
import com.android.gubonny.simplegithub.api.model.RepoSearchResponse;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

/**
 * Github API 저장소 검색 API 와
 * 저장소 정보 읽기 API 를 구현 함.
 */
public interface GithubApi {

    @GET("search/repositories")
    Call<RepoSearchResponse> searchRepository(@Query("q") String query);

    @GET("repos/{owner}/{name}")
    Call<GithubRepo> getRepository(
            @Path("owner") String ownerLogin,
            @Path("name") String repoName);
}

File: app/src/main/kotlin/com/android/gubonny/simplegithub/api/GithubApi.kt
package com.android.gubonny.simplegithub.api

import com.android.gubonny.simplegithub.api.model.GithubRepo
import com.android.gubonny.simplegithub.api.model.RepoSearchResponse

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

/**
 * Github API 저장소 검색 API 와
 * 저장소 정보 읽기 API 를 구현 함.
 */
interface GithubApi {

    @GET("search/repositories")
    fun searchRepository(@Query("q") query: String): Call<RepoSearchResponse>

    @GET("repos/{owner}/{name}")
    fun getRepository(
            @Path("owner") ownerLogin: String,
            @Path("name") repoName: String): Call<GithubRepo>
}
